omnibus/files/private-chef-cookbooks/private-chef/templates/default/oc_erchef.config.erb

```diff
@@ -2,304 +2,351 @@
-                {rabbitmq_port, <%= node['private_chef']['rabbitmq']['node_port'] %> },
-                {rabbitmq_user, <<"<%= node['private_chef']['rabbitmq']['user'] %>">>},
-                {rabbitmq_password, <<"<%= node['private_chef']['rabbitmq']['password'] %>">>},
-                {rabbitmq_vhost, <<"<%= node['private_chef']['rabbitmq']['vhost'] %>">>},
-                {rabbitmq_exchange, <<"">>},
-                {search_provider, <%= node['private_chef']['opscode-erchef']['search_provider'] %>},
-                {search_queue_mode, <%= node['private_chef']['opscode-erchef']['search_queue_mode'] %>},
-                {search_batch_max_size, <%= node['private_chef']['opscode-erchef']['search_batch_max_size'] %>},
-                {search_batch_max_wait, <%= node['private_chef']['opscode-erchef']['search_batch_max_wait'] %>},
-                {solr_service, [{root_url, "<%= @helper.solr_url() %>"},
-                                {timeout, <%= @solr_timeout %>},
-                                {init_count, <%= @solr_http_init_count %>},
-                                {max_count, <%= @solr_http_max_count %>},
-                                {cull_interval, <%= @solr_http_cull_interval %>},
-                                {max_age, <%= @solr_http_max_age %>},
-                                {max_connection_duration, <%= @solr_http_max_connection_duration %>},
-                                {ibrowse_options, <%= @solr_ibrowse_options %>}]}
+    {ip_mode, [ <%=PrivateChef['use_ipv6'] ? "ipv6,ipv4" : "ipv4" %> ] },
+    {rabbitmq_host, "<%= node['private_chef']['rabbitmq']['vip'] %>"},
+    {rabbitmq_port, <%= node['private_chef']['rabbitmq']['node_port'] %> },
+    {rabbitmq_user, <<"<%= node['private_chef']['rabbitmq']['user'] %>">>},
+    {rabbitmq_password, <<"<%= node['private_chef']['rabbitmq']['password'] %>">>},
+    {rabbitmq_vhost, <<"<%= node['private_chef']['rabbitmq']['vhost'] %>">>},
+    {rabbitmq_exchange, <<"">>},
+    {search_provider, <%= node['private_chef']['opscode-erchef']['search_provider'] %>},
+    {search_queue_mode, <%= node['private_chef']['opscode-erchef']['search_queue_mode'] %>},
+    {search_batch_max_size, <%= node['private_chef']['opscode-erchef']['search_batch_max_size'] %>},
+    {search_batch_max_wait, <%= node['private_chef']['opscode-erchef']['search_batch_max_wait'] %>},
+    {solr_service, [
+      {root_url, "<%= @helper.solr_url() %>"},
+      {timeout, <%= @solr_timeout %>},
+      {init_count, <%= @solr_http_init_count %>},
+      {max_count, <%= @solr_http_max_count %>},
+      {cull_interval, <%= @solr_http_cull_interval %>},
+      {max_age, <%= @solr_http_max_age %>},
+      {max_connection_duration, <%= @solr_http_max_connection_duration %>},
+      {ibrowse_options, <%= @solr_ibrowse_options %>}
+    ]}
+  ]},
+
+  <% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token'] %>
```

Comment:
Per the RFC, the token is optional. Chef's commercial product that relies on Data Collector requires the token, but others' implementations may not.